### PR TITLE
Enhance SearchableFinder to find searchable classes outside of app/

### PR DESCRIPTION
**Pull request by mfullbrook**
Enhance SearchableFinder to find searchable classes outside of `app/` directory.

This change enables scout-extended to work with non-typical Laravel projects that have models stored in directories outside of `app/`.  For example a project that uses the [laravel-modules](https://github.com/nWidart/laravel-modules) library.

* `SearchableFinder` uses config (if present) in `config/scout.php`.  The existing behaviour has not changed.  I'm unsure if this is the best approach as it requires the end user to add config that is not present in the template `scout.php`.
* Tests modified to test the discovery of models located in the `laravel/modules` directory.  All tests passing.

### CODE REVIEW

File: composer.json
Comment (nunomaduro) [inline composer.json]: Can't we detect the `project_scan_dirs` and `project_namespaces` from this value?

Comment (mfullbrook) [reply composer.json]: Oh yeah, that seems obvious now!  

For the benefit of the tests we would need to fetch mappings from `autoload` and `autoload-dev`.  Are you aware of an API exposed by composer to read the active project's composer file?  When running the tests in this package the path to composer.json will be different to when vendored into a laravel project.

Comment (mfullbrook) [reply composer.json]: So i have a solution...
```
$composer = Composer\Factory::create(new Composer\IO\NullIO(), null,null);
$activeJsonPath = $composer->getConfig()->getConfigSource()->getName();
// $activeJsonPath == '/Users/mark/code/scout-extended/composer.json'
```

Comment (mfullbrook) [reply composer.json]: I've refactored by interrogating Composer - this enhancement is now zero configuration!

File: composer.json
Comment (nunomaduro) [inline composer.json]: Sorry, buddy. You should wait me to answer before coding. You don't accept new dependencies because this.

Laravel internally have a function get composer's contents. Use that one.

Comment (mfullbrook) [reply composer.json]: No worries - i was keen!

Out of interest, for a package maintainer, why is adding the dependency a bad idea when Laravel also has the same dependency?

`Illuminate\Support\Composer` is not an option as it is a facade to calling composer scripts. The composer path is not exposed as public method.   

Instead we are now using `base_path('composer.json')` and calling `json_decode`.  This is actually better than using `Composer\Composer` because we are definitely loading the laravel project's composer.json, not the scout-extended's.

Comment (nunomaduro) [reply composer.json]: > Out of interest, for a package maintainer, why is adding the dependency a bad idea when Laravel also has the same dependency?

I didn't know this. Anyway, seems way clean now.

File: src/Helpers/SearchableFinder.php
Comment (nunomaduro) [inline src/Helpers/SearchableFinder.php]: I would prefer that you remove this.

File: src/Helpers/SearchableFinder.php
Comment (nunomaduro) [inline src/Helpers/SearchableFinder.php]: I would prefer that you remove this.

File: src/Helpers/SearchableFinder.php
Comment (nunomaduro) [inline src/Helpers/SearchableFinder.php]: Can you get a `$sources` and a `$namespaces` from this method?

File: src/Helpers/SearchableFinder.php
Comment (nunomaduro) [inline src/Helpers/SearchableFinder.php]: Return `$sources` and a `$namespaces` from this method instead of setting it into attributes of the class.

**Comment by nunomaduro** [comment]
@mfullbrook Do you think you are able to work on this buddy?

**Comment by mfullbrook** [comment]
Hi @nunomaduro, so sorry to have neglected this for a ridiculous amount of time🙈.

Anyway, I've refactored and fixed the broken tests. But, I have ten identical errors that, for me, are also present on master.

```
9) Tests\Features\TransformersTest::testTransformMethod
Algolia\AlgoliaSearch\Exceptions\UnreachableException: Impossible to connect, please check your Algolia Application Id.
```

I stepped through the trace and can see that the factory methods in the tests are triggering the dispatch of an `UpdateJob`.

Master is passing on travis, but master is failing on my local.  My branch is getting the same errors on Travis.  Any thoughts or pointers would be really appreciated.

**Comment by jnbn** [comment]
Any updates on this? @mfullbrook @nunomaduro

**Comment by mfullbrook** [comment]
Hi @DevinCodes, I've rebased my changed on top of master.  The tests all pass on my local machine when using my own Algolia Application.  Would you be able to give guidance as to why the tests are all failing on CircleCI with `Algolia\AlgoliaSearch\Exceptions\BadRequestException: Invalid Application-ID or API key`. Would really appreciate it. 🙏

**Comment by DevinCodes** [comment]
Hey @mfullbrook ! Happy new year, and sorry for the delay in reply.

There's currently an issue with the CI for forks, so I had to rerun your tests manually: they are all green 🎉 
 I'll give the PR a final review today, I think we're good to go after that! I'll keep you posted.

**Review by DevinCodes — CHANGES_REQUESTED** [review]
It looks good to me overall, thank you for your work on this!

Could you please rebase the PR on the `master` branch one last time? This ensures your PR is compatible with the latest release too, which added support for PHP8. Once that's done, I'll re-run the tests, and if all is green I think it's good to go.

**Comment by mfullbrook** [comment]
@DevinCodes thanks! I'm using this branch in my own app and it's working great.  I've rebased, please re-run the tests when you can.

**Review by DevinCodes — APPROVED** [review]
Thank you for your work on this @mfullbrook ! I'll release a new version with this feature tomorrow 🙂

**Comment by DevinCodes** [comment]
This is now available in [`v1.12.0`](https://github.com/algolia/scout-extended/releases/tag/v1.12.0)! 🎉

**Pull request merged**
